SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestRequireHandler.cpp

```diff
@@ -371,31 +371,9 @@ void SpatialFunctionalTestRequireHandler::LogAndClearStepRequires()
 		return A.Order < B.Order;
 	});
 
-	ASpatialFunctionalTestFlowController* FlowController = OwnerTest->GetLocalFlowController();
-	if (FlowController)
-	{
-		const FString& WorkerName = FlowController->GetDisplayName();
-
-		for (const auto& Require : RequiresOrdered)
-		{
-			FString Msg;
-			if (Require.bPassed)
-			{
-				Msg = FString::Printf(TEXT("%s [Passed] %s : \"%s\""), *WorkerName, *Require.Msg, *Require.StatusMsg);
-				UE_VLOG(nullptr, LogSpatialGDKFunctionalTests, Display, TEXT("%s"), *Msg);
-				UE_LOG(LogSpatialGDKFunctionalTests, Display, TEXT("%s"), *Msg);
-			}
-			else
-			{
-				Msg = FString::Printf(TEXT("%s [Failed] %s : %s"), *WorkerName, *Require.Msg, *Require.StatusMsg);
-				UE_VLOG(nullptr, LogSpatialGDKFunctionalTests, Error, TEXT("%s"), *Msg);
-				UE_LOG(LogSpatialGDKFunctionalTests, Error, TEXT("%s"), *Msg);
-			}
-		}
-
-		NextOrder = 0;
-		Requires.Empty();
-	}
+	NextOrder = 0;
```

Comment:
I am very confused as what this function is supposed to achieve?

Reply:
Ok I get it now.
Not sure I am onboard with this pattern, it leaves Required in empty state without being logged, its super tightly coupled to the calling code.

Reply:
I agree with this, but I wasn't sure what's the best way to go about this.

So the reason I did it this way was because I wanted to keep the RequireHandler's Requires member private, so I decided to make this function as a getter and a "flush" for it to be handed to the Spatial Functional Test Function for logging.

The alternative to this would be to make Requires public and let the test actor handle the empty(), open to any suggestion on the best pattern to use here.

Reply:
Ok I am going suggest going about this a different way:

- Use `TSortedMap` with a custom` SortPredicate` that uses the `Order` variable to do its sorted `Add`'s. This will make sure that any entry added to it will be sorted, this removes the need for a sorting operation.
- Make a `Print` function `FSpatialFunctionalTestRequire` that outputs the formatted string needed (you will need to prepend `*WorkerName` yourself but that is ok).
- You can now loop over the `TSortedMap` and call the Print function on each `FSpatialFunctionalTestRequire` entry, prefixing it with `*WorkerName`.
- Make ASpatialFunctionalTest a `friend` class of SpatialFunctionalTestRequireHandler and `Empty()` Requires when done in the calling scope.

Reply:
Alternatively you can subclass `TSortedMap` and add a `PrintAndConsume` function that removes the element when its printed.

Reply:
Okay so investigating this led to the creation of [UNR-5966](https://improbableio.atlassian.net/browse/UNR-5966). I will close this conversation since I think solving the ticket will completely remove this map and refactor that part of the code.

I am sorry we have promised you @simonsarginson  that we are going to merge this in this PR, but apparently it is a bit out of scope afterwards, sorry 😂